lucene/core/src/test/org/apache/lucene/util/bkd/TestBKD.java

```diff
@@ -844,7 +845,28 @@ public void visit(int docID, byte[] packedValue) {
               hits.set(docID);
             }
 
-            @Override
+          @Override
+          public void visit(DocIdSetIterator iterator, byte[] packedValue) throws IOException {
+              if (random().nextBoolean()) {
+                //check the default method is correct
+                IntersectVisitor.super.visit(iterator, packedValue);
+              } else {
+                assertTrue(iterator.docID() == -1);
```

Comment:
Can you use assertEquals so that we get the current doc ID according to the iterator in case of failure?